### PR TITLE
Change range to work with all car types, make icon car-type dependant

### PR DESCRIPTION
**Pull request by prvakt**
Changed range to use value from driving_range.total_range_in_km instead of a value from charging (so it can be used by all engine types)

### CODE REVIEW

File: custom_components/myskoda/sensor.py [hunk 1]
```diff
@@ -176,9 +176,11 @@ class RemainingDistance(ChargingSensor):
 
     @property
     def native_value(self) -> int | float | None:  # noqa: D102
-        if status := self._status():
-            if status.battery.remaining_cruising_range_in_meters is not None:
-                return status.battery.remaining_cruising_range_in_meters / 1000
+        if range := self.vehicle.driving_range:
```
Comment (dvx76) [inline custom_components/myskoda/sensor.py:179]: nit: maybe we should rename this class to `Range` now? And update the doc-string accordingly.

**Review by dvx76 — COMMENTED** [review]
Great. No idea why we didn't use the range from the driving_range right away 🤷‍♂️ 

Minor comment + linter wants an extra newline (tip: install https://docs.astral.sh/ruff/ in your editor and it will auto-format things for you right away)

File: custom_components/myskoda/sensor.py (hunk 1 above)
Comment (WebSpider) [reply custom_components/myskoda/sensor.py:179]: Also its not a charging sensor any more, chargingsensors only show up on cars with charging capability

**Review by WebSpider — APPROVED** [review]
Looks good to me. As soon as you fix the merge conflict, we can go ahead and commit this. Thanks for your contribution!

**Review by WebSpider — APPROVED** [review]

**Review by WebSpider — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**